Source/WebCore/dom/Document.cpp

```diff
@@ -9549,6 +9554,34 @@ std::optional<uint64_t> Document::noiseInjectionHashSalt() const
     return page()->noiseInjectionHashSaltForDomain(RegistrableDomain { m_url });
 }
 
+ContentVisibilityDocumentState& Document::contentVisibilityDocumentState()
+{
+    if (!m_contentVisibilityDocumentState)
+        m_contentVisibilityDocumentState = makeUnique<ContentVisibilityDocumentState>();
+    return *m_contentVisibilityDocumentState;
+}
+
+bool Document::isObservingContentVisibilityTargets() const
+{
+    return m_contentVisibilityDocumentState && m_contentVisibilityDocumentState->hasObservationTargets();
+}
+
+void Document::updateRelevancyOfContentVisibilityElements()
+{
+    if (!isObservingContentVisibilityTargets())
+        return;
+    if (m_contentVisibilityDocumentState->updateRelevancyOfContentVisibilityElements(m_contentRelevancyStatusUpdate))
```

Comment:
How do you know m_contentVisibilityDocumentState is not null here?

Reply:
`isObservingContentVisibilityTargets` checks for it